### PR TITLE
教你用 go-wecahty 和 web 协议开发机器人

### PR DESCRIPTION
**Pull request by dchaofei**
Please make sure to check the following boxes creating your pull request, thank you very much!

- [x] Add Blog Header, including title, author, image, categories, tags, etc.  

    For example:  

    ```md
    ---
    title: Your Great Blog Title
    author: your_github_id
    categories: project
    tags:
      - python
      - padlocal
    image: /assets/2021/02-your-blog-slug/teaser.jpg
    ---

    One line bio

    Your blog post content Your blog post content
    Your blog post content Your blog post content
    Your blog post content Your blog post content
    ```

    > It will be easier to start drafting your post by using a existing post file as your template, for example, [this one](https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_posts/2021-01-14-wechaty-puppet-service.md)

- [x] Keep all filenames lowercase, and use `-` to connect words instead of space. e.g. `2017-10-06-wechat-pc-impactor.md` instead of `2017-10-06-WeChat PC Impactor.md`
- [x] Embed video/pdf from YouTube/Bilibili/local (if there's any) by using `{% include iframe.html src="URL_or_PATH" %}`. (Learn more about how to embed file/url in the post by reading this [blog post](https://wechaty.js.org/2020/08/24/add-video-to-wechaty-blog/)
- [x] Save photo to local before publishing, to a folder named by `${MONTH}-${YOUR_BLOG_SLUG}/` under `/assets/2021/` directory.
- [x] Select a beautiful and meaningful teaser image for your blog
- [x] Create your contributor profile (if you are a first time contributor)

    For excample:

    > jekyll/_contributors/lijiarui.md

    ```yaml
    ---
    name: Jiarui LI (李佳芮)
    site: https://lijiarui.github.io
    bio: 句子互动创始人 & CEO，微软人工智能最具价值专家 (AI MVP)
    avatar: /jekyll/assets/contributors/lijiarui/avatar.png
    email: rui@juzi.bot
    twitter: https://weibo.com/u/2175505900
    ---
    ```

  Here is the example file:

  - lijiarui.md: <https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_contributors/lijiarui.md>
  - It will shown like this on wechaty website: <https://wechaty.js.org/contributors/lijiarui/>

To learn more about the contribute guideline, visit: <https://wechaty.js.org/docs/contributor-program/>

### CODE REVIEW

**Review by huan — COMMENTED** [review]
Could you please add link of this great blog post link to the following docs?

https://wechaty.js.org/docs/polyglot/go/
https://wechaty.js.org/docs/puppet-providers/wechat

**Comment by dchaofei** [comment]
> Could you please add link of this great blog post link to the following docs?
> 
> https://wechaty.js.org/docs/polyglot/go/
> https://wechaty.js.org/docs/puppet-providers/wechat

已经加到了文章里，可以 review 了

**Comment by dchaofei** [comment]
ci 终于变绿了......

File: jekyll/_posts/2021-04-16-go-wechaty-use-web.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+`WECHATY_PUPPET_SERVER_PORT` 环境变量是我们启动 web 协议 puppet service 提供的端口
+
+`WECHATY_TOKEN` 是我们随机生成的字符串，用于把当前puppet service 注册到 puppet 网关
+
+二、启动 go-wecahty 写的一个简单的 ding-dong 机器人
+
+```bash
+git clone https://github.com/wechaty/go-wechaty-getting-started.git
+
+WECHATY_PUPPET_SERVICE_ENDPOINT=127.0.0.1:30001 make bot
```
Comment (doylecnn) [inline jekyll/_posts/2021-04-16-go-wechaty-use-web.md:67]: 顺便写一下可以这样不依赖环境变量：
```go
var bot = wechaty.NewWechaty(wechaty.WithPuppetOption(wp.Option{
        Endpoint: "127.0.0.1:30001",
        Token: "3d415ebb-7a6f-4cba-b602-1f4ae400f011",
        Timeout: time.Duration(2*time.Minute),
}))
```

**Review by doylecnn — APPROVED** [review]

File: jekyll/_posts/2021-04-16-go-wechaty-use-web.md (hunk 1 above)
Comment (dchaofei) [reply jekyll/_posts/2021-04-16-go-wechaty-use-web.md:67]: 加上了

**Review by huan — APPROVED** [review]
LGTM

**Comment by huan** [comment]
Thank your very much, merged.

Could you please modify the following two docs page:

https://wechaty.js.org/docs/polyglot/go/
https://wechaty.js.org/docs/puppet-providers/wechat

To add this blog post link to them?

**Pull request merged**